packages/amplify_auth_cognito/android/src/main/kotlin/com/amazonaws/amplify/amplify_auth_cognito/AuthCognitoHubEventStreamHandler.kt

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amazonaws.amplify.amplify_datastore
+
+import android.os.Handler
+import android.os.Looper
+import com.amplifyframework.auth.AuthChannelEventName
+import com.amplifyframework.core.Amplify
+import com.amplifyframework.core.InitializationStatus
+import com.amplifyframework.hub.HubChannel
+import com.amplifyframework.hub.HubEvent
+import com.amplifyframework.hub.SubscriptionToken
+import io.flutter.plugin.common.EventChannel
+import java.util.concurrent.CountDownLatch
+
+
+class AuthCognitoHubEventStreamHandler : EventChannel.StreamHandler {
+    private var eventSink: EventChannel.EventSink? = null
+    private val handler = Handler(Looper.getMainLooper())
+    private lateinit var token: SubscriptionToken
+    private val LOG = Amplify.Logging.forNamespace("amplify:flutter:auth_cognito")
+    private var forwardHubResponse : (event: Map<String, Any>) -> Unit
+
+    constructor(){
+        forwardHubResponse = {event: Map<String, Any> -> handler.post {
+            eventSink?.success(event)
+        }}
+    }
+
+    constructor(latch: CountDownLatch) {
+        forwardHubResponse = { latch.countDown() }
+    }
+
+    override fun onListen(argunents: Any?, sink: EventChannel.EventSink) {
+        eventSink = sink
+        token = getHubListener()
+    }
+
+     fun getHubListener(): SubscriptionToken {
```

Comment:
I preferred `getHubListener' to denote that we are actually returning the subscription token.

Reply:
a listener usually signifies a `callback` method, but in this case you are creating/setting/subscribing to a subscription and returning a token used for canceling it later. Since it's an internal method, it's not critical to change.